src/Common/src/System/Security/Cryptography/DSAOpenSsl.cs

```diff
@@ -233,6 +229,18 @@ public override bool VerifySignature(byte[] hash, byte[] signature)
                 return Interop.Crypto.DsaVerify(key, hash, hash.Length, openSslFormat, openSslFormat.Length);
             }
 
+            private void SetKey(SafeDsaHandle newKey)
+            {
+                // Use ForceSet instead of the property setter to ensure that LegalKeySizes doesn't interfere
+                // with the already loaded key.
+                ForceSetKeySize(BitsPerByte * Interop.Crypto.DsaKeySize(newKey));
+
+                _key = new Lazy<SafeDsaHandle>(() => newKey);
```

Comment:
You could do:
```C#
var key = new Lazy<SafeDsaHandle>(() => newKey, threadSafe: false);
var dummy = key.Value;
_key = key;
```
That way we don't pay the double-checked locking costs on initializing the lazy instance.